Tests/interop/net/field/test_instance_fields.py

```diff
@@ -235,26 +236,27 @@ def f22(): t.InstanceSimpleClassField.__set__(v, SimpleClass(540))
         def f23(): t.InstanceSimpleGenericClassField.__set__(v, SimpleGenericClass[str]("string".upper()))
         def f24(): t.InstanceSimpleInterfaceField.__set__(v, ClassImplementSimpleInterface(78))
 
-        funcs = [ eval("f%s" % i) for i in range(1, 25) ]
-        if clr.GetClrType(t).IsValueType:
-            for f in funcs:
-                self.assertWarns(RuntimeWarning, 'may result in updating a copy', f)
-        else: 
-            for f in funcs: f()
-            self._test_verify(o)
-            
-            # set values which need conversion.
-            t.InstanceInt16Field.__set__(v, 100);        self.assertEqual(o.InstanceInt16Field, 100)
-            t.InstanceBooleanField.__set__(v, 0);         self.assertEqual(o.InstanceBooleanField, False)
+        for i in range(1, 25):
+            f = eval("f%s" % i)
+            if clr.GetClrType(t).IsValueType:
+                    self.assertWarnsRegex(RuntimeWarning, 'may result in updating a copy', f)
+            else:
+                f()
+
+        self._test_verify(o)
 
-            # set bad values 
-            def f1(): t.InstanceInt16Field.__set__(v, "abc")
-            def f2(): t.InstanceCharField.__set__(v, "abc")
-            def f3(): t.InstanceEnumField.__set__(v, EnumInt32.B)
+        # set values which need conversion.
+        t.InstanceInt16Field.__set__(v, 100);        self.assertEqual(o.InstanceInt16Field, 100)
```

Comment:
This should be `t.InstanceInt16Field.__set__(v, long(100))`. Looks like the fact it was a `long` somehow got lost when it got ported from IronPython2.

Reply:
Done